views/audit/bigquery_audit_log_v2.sql

```diff
@@ -0,0 +1,1161 @@
+            "/")[SAFE_OFFSET(1)],
+      ":",
+      SPLIT(
+        JSON_EXTRACT_SCALAR(protopayload_auditlog.metadataJson,
+          '$.routineDeletion.jobName'),
+            "/")[SAFE_OFFSET(3)]
+    ) AS jobId,
+  FROM `project_id.dataset_id.cloudaudit_googleapis_com_data_access`
+) -- End of WITH clauses
+SELECT
+  principalEmail AS principalEmail,
+  callerIp,
+  serviceName,
+  methodName,
+  eventName,
+  projectId,
+  jobId,
+  IF(jobChangeJobName IS NULL, False, True) AS hasJobChangeEvent,
+  IF(tableChangeJobName IS NULL, False, True) AS hasTableChangeEvent,
+  IF(tableCreationJobName IS NULL, False, True) AS hasTableCreationEvent,
+  IF(tableDeletionJobName IS NULL, False, True) AS hasTableDeletionEvent,
+  IF(tableDataReadJobName IS NULL, False, True) AS hasTableDataReadEvent,
+  IF(tableDataChangeJobName IS NULL, False, True) AS hasTableDataChangeEvent,
+  IF(modelDeletionJobName IS NULL, False, True) AS hasModelDeletionEvent,
+  IF(modelCreationJobName IS NULL, False, True) AS hasModelCreationEvent,
+  IF(modelMetadataChangeJobName IS NULL, False, True) AS hasModelMetadataChangeEvent,
+  IF(routineCreationJobName IS NULL, False, True) AS hasRoutineCreationEvent,
+  IF(routineChangeJobName IS NULL, False, True) AS hasRoutineChangeEvent,
+  IF(routineDeletionJobName IS NULL, False, True) AS hasRoutineDeletionEvent,
+  STRUCT(
+    EXTRACT(MINUTE FROM jobStatsStartTime) AS minuteOfDay,
+    EXTRACT(HOUR FROM jobStatsStartTime) AS hourOfDay,
+    EXTRACT(DAYOFWEEK FROM jobStatsStartTime) - 1 AS dayOfWeek,
+    EXTRACT(DAYOFYEAR FROM jobStatsStartTime) AS dayOfYear,
+    EXTRACT(WEEK FROM jobStatsStartTime) AS week,
+    EXTRACT(MONTH FROM jobStatsStartTime) AS month,
+    EXTRACT(QUARTER FROM jobStatsStartTime) AS quarter,
+    EXTRACT(YEAR FROM jobStatsStartTime) AS year
+  ) AS jobStartDate,
+  jobStatsRuntimeMs AS jobRuntimeMs,
+  jobStatsRuntimeSecs AS jobRuntimeSec,
+  REGEXP_CONTAINS(jobId, 'beam') AS isBeamJob,
+  REGEXP_CONTAINS(queryConfigQuery, 'cloudaudit_googleapis_com_data_access') AS isAuditDashboardQuery,
+  jobStatsTotalSlotMs / jobStatsRuntimeMs AS avgSlots,
+  /*
+   * The following statement breaks down the query into minute buckets
+   * and provides the average slot usage within that minute. This is a
+   * crude way of making it so you can retrieve the average slot utilization
+   * for a particular minute across multiple queries.
+   */
+  ARRAY(
+    SELECT
+      STRUCT(
+        TIMESTAMP_TRUNC(
+          TIMESTAMP_ADD(jobStatsStartTime, INTERVAL bucket_num MINUTE), MINUTE
+        ) AS time,
+        jobStatsTotalSlotMs / jobStatsRuntimeMs AS avgSlotUsage
+      )
+    FROM UNNEST(GENERATE_ARRAY(1, jobStatsExecutionMinuteBuckets)) AS bucket_num
+  ) AS jobExecutionTimeline,
+  ARRAY_LENGTH(queryJobStatsReferencedTables) AS totalTablesProcessed,
+  ARRAY_LENGTH(queryJobStatsReferencedViews) AS totalViewsProcessed,
+  (queryJobStatsTotalBilledBytes / pow(2,30)) AS totalBilledGigabytes,
```

Comment:
Why don't we also include `totalBilledBytes`?

Reply:
It's included in the jobStats struct